--- src/Makefile	2024-01-12 17:25:58.000000000 +0100
+++ src/Makefile.new	2025-01-24 11:57:11.000000000 +0100
@@ -55,7 +55,7 @@
 ifndef FORTRAN
   # GNU Fortran as the default Fortran compiler
   # NOTE: use gfortran version 9 or greater, or any compiler compatible with Fortran 2008.
-  FORTRAN=gfortran
+  FORTRAN=ftn
   
   # Comment the following line to disable parallelization with OpenMP
   OPENMP=-DOPENMP -fopenmp
@@ -80,7 +80,7 @@
 ifndef LAPACK
   # Path to the LAPACK library
   # If it is installed system-wide just use "-llapack"
-  LAPACK=-llapack
+#  LAPACK=-llapack
 endif
 
 # DEFINITION OF PATHS:
