a workaround to avoid https://github.com/rstudio/rstudio/issues/16675
author: Karl Johan Westrin (PDC KTH)
diff -ruN rstudio-2024.12.0-467.orig/src/cpp/session/modules/environment/SessionEnvironment.cpp rstudio-2024.12.0-467/src/cpp/session/modules/environment/SessionEnvironment.cpp
--- rstudio-2024.12.0-467.orig/src/cpp/session/modules/environment/SessionEnvironment.cpp	2026-03-10 19:38:09.000000000 +0100
+++ rstudio-2024.12.0-467/src/cpp/session/modules/environment/SessionEnvironment.cpp	2026-03-11 09:22:19.000000000 +0100
@@ -22,8 +22,6 @@
 
 #include <algorithm>
 
-#include <boost/container/flat_map.hpp>
-
 #include <core/Exec.hpp>
 #include <core/RecursionGuard.hpp>
 #include <core/system/LibraryLoader.hpp>
@@ -91,10 +89,6 @@
 // whether or not the global environment can safely be serialized
 bool s_isGlobalEnvironmentSerializable = true;
 
-// whether or not particular objects can be safely serialized
-using SerializationCache = boost::container::flat_map<SEXP, bool>;
-SerializationCache s_serializationCache;
-
 // by default, use regular 'INTEGER' accessor; 'INTEGER_OR_NULL' will be loaded
 // if provided by this version of R
 int* (*INTEGER_OR_NULL)(SEXP) = INTEGER;
@@ -206,12 +200,6 @@
 
 bool isGlobalEnvironmentSerializable()
 {
-   // Start building a new cache of serialized object state.
-   SerializationCache newCache;
-   
-   // Flag tracking whether we found an object which cannot be serialized.
-   bool allValuesSerializable = true;
-   
    // Iterate over values in the global environment, and compute whether they can be serialized.
    SEXP hashTableSEXP = HASHTAB(R_GlobalEnv);
    R_xlen_t n = Rf_xlength(hashTableSEXP);
@@ -221,29 +209,16 @@
       for (; frameSEXP != R_NilValue; frameSEXP = CDR(frameSEXP))
       {
          // Compute whether the value can be serialized.
-         // If we already have a cached value from a prior computation, use it.
-         //
-         // TODO: If an R environment is updated in-place with a value that
-         // makes it no longer serializable, we would fail to detect this.
-         // Is this okay? Or should we avoid caching results for environments?
          if (!r::internal::isImmediateBinding(frameSEXP))
          {
             SEXP valueSEXP = CAR(frameSEXP);
-            bool canBeSerialized = s_serializationCache.contains(valueSEXP)
-                  ? s_serializationCache.at(valueSEXP)
-                  : isSerializable(valueSEXP);
-            newCache[valueSEXP] = canBeSerialized;
-            allValuesSerializable = allValuesSerializable && canBeSerialized;
+            bool canBeSerialized = isSerializable(valueSEXP);
+            if (!canBeSerialized) { return false; }
          }
       }
    }
-   
-   // Set the new cache.
-   s_serializationCache.clear();
-   s_serializationCache = newCache;
-   
    // Return true only if all values can be serialized.
-   return allValuesSerializable;
+   return true;
 }
 
 bool isValidSrcref(SEXP srcref)
